casper/src/main/rholang/BasicWallet.rho

```diff
@@ -0,0 +1,65 @@
+//scalapackage coop.rchain.rholang.wallet
+
+//assumes standard purse API from MakeMint
+contract @"BasicWallet"(@purse, @algorithm, @pk, return) = {
+  new cryptoVerifyCh in {
+    match algorithm {
+      "ed25519" => {
+        cryptoVerifyCh!(["ed25519Verify"])
+      }
+      _ => {
+        cryptoVerifyCh!([])
+      }
+    } |
+    for(@maybeCryptoVerify <- cryptoVerifyCh) {
+      match maybeCryptoVerify {
+        []             => { return!([]) }
+        [cryptoVerify] => {
```

Comment:
Can this be written as
```
if(algorithm == "ed25519Verify") {
  …
} else { return!([]) }
```

Reply:
But in the future it won't be that simple. The present design easily extends to including other algorithms. In particular we will be include `secp256k1` as soon as you're done adding it to Rholang.